standards/application/messaging-api.md

```diff
@@ -0,0 +1,944 @@
+
+##### `staticStoreNode`
+A list of `Multiaddr` addresses to remote peers that MUST be used for retrieving past messages or performing infrequent queries using the [STORE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/13/store.md) protocol.
+If not provided, the implementation SHOULD utilize either `preferredServiceNodes` or nodes discovered via the network.
+
+##### `preferredServiceNodes`
+A list of `Multiaddr` addresses to remote peers that SHOULD be used for obtaining various network resources.
+
+These resources MAY include:
+- infrequent [STORE](../standards/core/store.md) queries;
+- a [LIGHTPUSH](../standards/core/lightpush.md) endpoint for broadcasting data;
+- a [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) endpoint for receiving circulating data.
+
+If preferred nodes are not provided or are unreachable (i.e., a connection cannot be established due to errors or timeouts),
+the implementation SHOULD use nodes discovered through the network.
+
+##### `bootstrapNodes`
+This property MUST be provided.
+
+A list of `Multiaddr` addresses to remote peers that MUST be used for any applicable method of discovery that MAY include:
+- [DNS Discovery](https://eips.ethereum.org/EIPS/eip-1459);
+- [DISCV5](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/33/discv5.md);
+- [PEER-EXCHANGE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/34/peer-exchange.md).
+
+##### `storeConfirmation`
+This optional property defaults to `true`.
+- If set to `true`, the `stored` field on `MessageRecord` MUST be populated, and implementations MUST employ store‑based reliability as specified in [P2P‑RELIABILITY](./p2p‑reliability.md).
+- If set to `false`, the `stored` field on `MessageRecord` MUST NOT be populated.
+For more information about the [STORE](../standards/core/store.md) query and the `MessageRecord`, refer to the `Message Storage API` section.
+
+##### `receiveConfirmation`
+This optional property defaults to `true`.
+- If set to `true`, the node MUST initiate message‑reception detection as specified in [P2P‑RELIABILITY](./p2p‑reliability.md) and populate the `received` field on `MessageRecord`.  
+- If set to `false`, the `received` field MUST NOT be populated, unless the `Subscribe API` is invoked.
+
+##### `confirmContentTopics`
+An optional property that SHOULD be provided in conjunction with either `storeConfirmation` or `receiveConfirmation`.
+It includes an array of `contentTopics` that the node MUST start monitoring in the background upon initialization,
+as defined in the [TOPICS](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/informational/23/topics.md#content-topic-format) specification.
```

Comment:
I'd remove that. I assume this because you want to be able to setup some filter subscription for content topics the user will send to?
We are making the API more complex here. It seems that filter confirmations are unpractical?